docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,64 @@
+Etcher Architecture
+===================
+
+This document aims to serve as a high level overview of how Etcher works, specially oriented for contributors who want to understand the big picture.
+
+Module architecture
+-------------------
+
+Instead of embedding all the functionality required to create a full-featured image writer as a monolithic project, we try to hard to follow the ["lego block approach"](https://github.com/sindresorhus/ama/issues/10#issuecomment-117766328).
+
+This has the advantage of allowing other applications to re-use logic we implemented for Etcher in their own project, even for things we didn't expect, which leads to users benefitting from what we've built, and we benefitting from user's bug reports, suggestions, etc, as an indirect way to make Etcher better.
+
+The fact that low level details are scattered around many different modules makes it a bit hard for new contributor to wrap their heads around the project as a whole, and get a clear high level view of how things work or where to submit their work or bug reports.
```

Comment:
makes it a bit hard for new contributor -> can make it challenging for a new contributor